File: Visual C++/LearningCpp/CountingSort.cpp
#include<iostream>

using namespace std;

class CountingSort
{
public:
	static int* sort(int arr[],int max, int n)
	{
		int *C = new int[max + 1];
		int *B = new int[n];

		for (int i = 0;i < max + 1;i++)
		{
			C[i] = 0;
		}

		for (int i = 0;i < n;i++)
		{
			C[arr[i]]++;
		}

		for (int i = 1;i < max + 1;i++)
		{
			C[i] = C[i] + C[i - 1];
		}

		for (int i = n - 1;i >= 0;i--)
		{
			B[C[arr[i]] - 1] = arr[i];
			C[arr[i]]--;
		}

		return B;
	}

	
};

int main()
{
	int arr[] = { 12,34,1,2,6,78,6,6,2,1 };

	cout << "Original array : \n\n";
	for (int i = 0;i < 10;i++)
	{
		cout << " " << arr[i];
	}

	int *temp = CountingSort::sort(arr, 78, 10);

	cout << "\nArray after sorting : \n";
	for (int i = 0;i < 10;i++)
	{
		cout << " " << temp[i];
	}


}

File: Javascript/CountingSort.js
function sort(arr,M)
{
    var A = [];
    for(var i=0;i<M;i++)
    {
        A[i]=0;
    }

    for(var i=0;i<arr.length;i++)
    {
        A[arr[i]]++;
    }

    for(var i=1;i<M;i++)
    {
        A[i]=A[i-1]+A[i];
    }

    var C = [];

    for(var i=arr.length-1;i>=0;i--)
    {
        C[A[arr[i]]-1]=arr[i];
        A[arr[i]]--;
    }

    return C;

}

var arr = [12,23,45,11,2,2,5,23];
var arr1 = sort(arr,46);
console.log(arr1);